cairo/ethereum/cancun/fork.cairo

```diff
@@ -1341,3 +1345,99 @@ func _process_withdrawals_inner{
 
     return _process_withdrawals_inner{state=state, trie=trie}(index + 1, withdrawals);
 }
+
+func state_transition{
+    range_check_ptr,
+    range_check96_ptr: felt*,
+    add_mod_ptr: ModBuiltin*,
+    mul_mod_ptr: ModBuiltin*,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    chain: BlockChain,
+}(block: Block) {
+    alloc_locals;
+
+    let parent_header = chain.value.blocks.value.data[
+        chain.value.blocks.value.len - 1
+    ].value.header;
+
+    let excess_blob_gas = calculate_excess_blob_gas(parent_header);
+    with_attr error_message("InvalidBlock") {
+        assert block.value.header.value.excess_blob_gas = excess_blob_gas;
+    }
+
+    validate_header(block.value.header, parent_header);
+
+    with_attr error_message("InvalidBlock") {
+        assert block.value.ommers.value.len = 0;
+    }
+
+    let state = chain.value.state;
+    let block_hashes = get_last_256_block_hashes(chain);
+    let output = apply_body{state=state}(
+        block_hashes,
+        block.value.header.value.coinbase,
+        block.value.header.value.number,
+        block.value.header.value.base_fee_per_gas,
+        block.value.header.value.gas_limit,
+        block.value.header.value.timestamp,
+        block.value.header.value.prev_randao,
+        block.value.transactions,
+        chain.value.chain_id,
+        block.value.withdrawals,
+        block.value.header.value.parent_beacon_block_root,
+        excess_blob_gas,
+    );
```

Comment:
You need to re-bind state inside `chain` after "moving" it to `apply_body`

Reply:
Thank you!
that was the problem 
fixed